### PR TITLE
Improve contributing, publishing and CICD docs

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

Docs :)

### CODE REVIEW

File: docs/CICD.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# CICD
+
+## TLDR (day-to-day dev flow)
+
+Every time a PR is merged into main, an automated staging relase will happen. Once the packages are pushed to `npm`, the following actions will take place:
```
Comment (desiprisg) [inline docs/CICD.md:5]: Let's do "into the main branch".

File: docs/CICD.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Actions that will be triggered:
+
+- `clerkinc/cloudflare-workers`: The latest clerk-js versions in `clerkjs-proxy/wrangler.toml` will be updated an a PR will open. Follow the instructions in the PR to manually release a new `clerkjs-proxy` worker.
+
+For more details, refer to [PUBLISH.md](https://github.com/clerkinc/javascript/blob/main/docs/PUBLISH.md).
+
+## Automated staging releases
+
+A staging release will be triggered every time PR is merged into main. Once versioning and publishing is done, the `clerkinc/javascript` repo will dispatch a workflow event, notifying other related Clerk repos of the new releases.
```
Comment (desiprisg) [inline docs/CICD.md:22]: "into the main branch" as well ?!

**Review by desiprisg — APPROVED** [review]

File: docs/PUBLISH.md [hunk 3]
```diff
@@ -4,58 +4,38 @@ _Note: Only core maintainers can publish packages._
...
+To release a new stable version of all Clerk packages, find the "Version Packages" PR, verify the changes, and merge it.
 
-After all the features have been merged and we want to create a new release, we use `npm run bump`.
+## Publishing staging package versions (`@staging`)
 
-This script will use `lerna version` to check which packages need to be updated and in what way based on the updates since the previous release.
+An automated staging release will be take place every time a PR gets merged into main.
 
-The command will guide you through the version changes that are detected and will:
+- Snapshot versions use the following format: `@clerk/package@x.y.z-staging.commit`, where `package` is the package name, `x`,`y`,`z` are the major, minor and patch versions respectively, `staging` is a stable prerelease mame and `commit` is the id of the last commit in the branch.
```
Comment (panteliselef) [inline docs/PUBLISH.md:19]:
```suggestion
- Staging ...
```

File: docs/PUBLISH.md [hunk 4]
```diff
@@ -4,58 +4,38 @@ _Note: Only core maintainers can publish packages._
...
+- Currently, staging version changes are _not_ committed to the repo and no git tags will be generated. Using this strategy, we avoid merge conflicts, allowing us to constantly deploy staging versions without switching the repo to a "prerelease" mode.
+- During a snapshot release, `@clerk/clerk-js` will also be released. If needed, use the `clerkJSVersion` prop to use a specific version, eg: `<ClerkProvider clerkJSVersion='@clerk/clerk-js@4.1.1-staging.90012' />`
```
Comment (panteliselef) [inline docs/PUBLISH.md:21]:
```suggestion
- During a staging release, `@clerk/clerk-js` will also be released. If needed, use the `clerkJSVersion` prop to use a specific version, eg: `<ClerkProvider clerkJSVersion='@clerk/clerk-js@4.1.1-staging.90012' />`
```

**Review by panteliselef — APPROVED** [review]

**Review by anagstef — APPROVED** [review]

File: docs/PUBLISH.md (hunk 4 above)
Comment (nikosdouvlis) [reply docs/PUBLISH.md:21]: Thanks Panteli :)

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**